packages/core/test/Synchronizer.test.ts

```diff
@@ -0,0 +1,639 @@
+                epochKey: reputationProof.epochKey,
+                epoch: Number(reputationProof.epoch),
+                proveMinRep: minPosRep ? true : false,
+                minRep: minPosRep,
+                status: 1,
+            },
+        })
+        expect(comment).not.to.be.null
+        const post = await db.findOne('Post', {
+            where: {
+                transactionHash: postId,
+                commentCount: 1,
+            },
+        })
+        expect(post).not.to.be.null
+        const record = db.findOne('Record', {
+            where: {
+                to: reputationProof.epochKey,
+                from: reputationProof.epochKey,
+                upvote: 0,
+                downvote: defaultCommentReputation,
+                epoch: Number(reputationProof.epoch),
+                action: ActionType.Comment,
+                data: transactionHash,
+                transactionHash,
+            },
+        })
+        expect(record).not.to.be.null
+        const epkRecord = db.findOne('EpkRecord', {
+            where: {
+                epk: reputationProof.epochKey,
+                epoch: Number(reputationProof.epoch),
+                spent: defaultCommentReputation,
+                posRep: 0,
+                negRep: 0,
+            },
+        })
+        expect(epkRecord).not.to.be.null
+    })
+
+    it('submit comment should succeed and update db (if data saved before)', async () => {
+        const attesterId = BigInt(
+            await unirepContract.attesters(unirepSocialContract.address)
+        )
+        const { transactionHash: postId } = await publishPost(
+            unirepSocialContract,
+            ethers.provider
+        )
+        const id = new ZkIdentity()
+        await unirepSocialContract
+            .userSignUp(id.genIdentityCommitment())
+            .then((t) => t.wait())
+        const userState = await genUserState(
+            ethers.provider,
+            unirepContract.address,
+            id
+        )
+        const proveGraffiti = BigInt(0)
+        const minPosRep = 20,
+            graffitiPreImage = BigInt(0)
+        const epkNonce = 0
+        const defaultCommentReputation = 3
+        const reputationProof = await userState.genProveReputationProof(
+            attesterId,
+            epkNonce,
+            minPosRep,
+            proveGraffiti,
+            graffitiPreImage,
+            defaultCommentReputation
+        )
+
+        const content = 'some comment text'
+        const hashedContent = ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes(content)
+        )
+
+        const { transactionHash } = await unirepSocialContract
+            .leaveComment(
+                postId,
+                hashedContent,
+                reputationProof.publicSignals,
+                reputationProof.proof,
+                { value: attestingFee }
+            )
+            .then((t) => t.wait())
+
+        await db.create('Comment', {
```

Comment:
This is a race condition, you're waiting for the transaction above so it will likely be processed by the synchronizer before this create call is hit.

Try calling `t.wait()` after `db.create`